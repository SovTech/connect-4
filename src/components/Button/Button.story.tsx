import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { Button, Card } from '../index';
import { action } from '@storybook/addon-actions';

storiesOf('Button', module)
  .add('Regular', () => (
    <Card
      title='Buttons'
    >
      <Button
        onClick={action('onClick')}
        text='Nothing to see here'
      />
    </Card>
  ))
  .add('Disabled', () => (
    <Card
      title='Buttons'
    >
      <Button
        onClick={action('onClick')}
        disabled={true}
        text='This button is disabled'
      />
    </Card>
  ))
  .add('Loading', () => (
    <Card
      title='Buttons'
    >
      <Button
        onClick={action('onClick')}
        isLoading={true}
        text='This button is loading'
      />
    </Card>
  ))
  .add('Icon', () => (
    <Card
      title='Buttons'
    >
      <Button
        onClick={action('onClick')}
        text='Icon Button'
      />
    </Card>
  ))
  .add('Long text', () => (
    <Card
      title='Buttons'
    >
      <Button
        onClick={action('onClick')}
        text='This is a button with some really long text.
                 Its going to mess things up on purpose.
                  Nothing to see here. Ignore all of this.'
      />
    </Card>
  ));
