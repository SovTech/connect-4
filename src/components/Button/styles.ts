import theme from '../../theme';
import styled from 'styled-components';
import { IS_TEST } from '../../constants';
import { ButtonProps } from './index';

export const Button = styled<ButtonProps, 'button'>('button')`
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  text-decoration: none;
  border-radius: ${theme.borderRadius};
  margin: ${props => props.noMargin ? 0 : theme.marginQuarter};
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  box-shadow: 0 10px 20px 0 rgba(0, 0, 0, 0.08);
  cursor: pointer;
  outline: 0;
  font-size: ${theme.h4Size};
  font-weight: 500;
  color: ${theme.textColor};
  transition: all 0.3s ease;
  border: 0;
  border-bottom: thick solid ${theme.overlayColorLight};
  overflow: hidden;
  padding: ${theme.marginHalf} ${theme.margin};
  width: 100%;
  background-color: ${IS_TEST ? theme.testBackgroundColor : theme.baseColorDarker20};
     
  &:hover {
    box-shadow: 0 15px 20px 0 rgba(0, 0, 0, 0.15);
  }

  &:active {
    background-color: ${IS_TEST ? theme.testBackgroundColor : theme.baseColorDarker20};
  }
`;

export const Icon = styled.div`
  padding-right: ${theme.marginHalf};
  display: flex;
  align-items: center;
  justify-content: center;

  * {
    stroke: ${theme.inverseTextColor};
  }
`;