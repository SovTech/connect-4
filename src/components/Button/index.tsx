// Button

import * as React from 'react';
import { Button, Icon } from './styles';

export type ButtonProps = {
  disabled?: boolean,
  isLoading?: boolean,
  noMargin?: boolean,
  text?: string,
  title?: string,
  className?: string,
  icon?: Node,
  onClick: Function | any,
}

export default (props: ButtonProps) => {
  const disabled = props.isLoading || props.disabled;
  return (
    <Button
      title={props.title}
      noMargin={props.noMargin}
      disabled={props.disabled}
      onClick={disabled ? null : props.onClick}
    >
      {props.icon ? <Icon>{props.icon}</Icon> : null}
      {props.isLoading ? 'Loading...' : props.text}
    </Button>
  );
}
