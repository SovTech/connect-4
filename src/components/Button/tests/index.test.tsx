import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Button from '../index';

describe('Button Component', () => {
  it('renders a Button correctly', () => {
    const tree = renderer
      .create(<Button text='Button Title' onClick={() => console.log('onClick')} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a disabled Button correctly', () => {
    const tree = renderer
      .create(<Button disabled={true} text='Button Title' onClick={() => console.log('onClick')} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a loading Button correctly', () => {
    const tree = renderer
      .create(<Button isLoading={true} text='Button Title' onClick={() => console.log('onClick')} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
