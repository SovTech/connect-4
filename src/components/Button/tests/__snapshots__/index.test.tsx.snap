// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders a Button correctly 1`] = `
<button
  className="sc-bdVaJa gXrLnl"
  disabled={undefined}
  onClick={[Function]}
  title={undefined}
>
  Button Title
</button>
`;

exports[`Button Component renders a disabled Button correctly 1`] = `
<button
  className="sc-bdVaJa gXrLnl"
  disabled={true}
  onClick={null}
  title={undefined}
>
  Button Title
</button>
`;

exports[`Button Component renders a loading Button correctly 1`] = `
<button
  className="sc-bdVaJa gXrLnl"
  disabled={undefined}
  onClick={null}
  title={undefined}
>
  Loading...
</button>
`;
